src/haz3lcore/statics/Ctx.re

```diff
@@ -67,6 +67,25 @@ let free_in = (ctx_before: t, ctx_after, free: co): co => {
   );
 };
 
+let is_unused_in_free = (free: co, name: string): option(bool) => {
```

Comment:
a co_ctx is a VarMap so you can use VarMap.lookup. This fn should probably just be a one-liner,something like
let co_ctx_contains = (free, name): bool =>  VarMap.lookup(free,name) !=None, and then at the call site you not it to get the opposite